### PR TITLE
Typical history cleanup

### LINKED ISSUE #5: Cleaning up the Typical History postprocessor
Originally it was believed there was a bug in the TypicalHistory postprocessor; however, it was discovered that its operation is merely obfuscated by some coding choices.

For example, there are literally 0 comments in the postprocessor (aside from function definitions), and variable names such as "tempData" and "tempInp" don't give any clarity to what actions are being performed.

It would be immeasurably helpful in the future to add comments and rename variables so that we can debug/make use of this postprocessor more in the future.

Similarly, the documentation for this postprocessor could benefit from some explanatory examples.

### PR DESCRIPTION
**Pull request by PaulTalbot-INL**
Will close #5.

Renames variables, adds comments, clarifies documentation in both the code and the user manual for the Typical History PostProcessor.

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
The *.md files were unintentionally included; this should be merged AFTER #3.

**Comment by joshua-cogliati-inl** [comment]
Can this just be merged with #4  ?

**Comment by PaulTalbot-INL** [comment]
Yes, if #4 is approved than these others that include it could be merged and take care of it in one fell swoop.

File: framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py [hunk 1]
```diff
@@ -41,8 +43,22 @@ def initialize(self):
     PostProcessorInterfaceBase.initialize(self)
     self.inputFormat  = 'HistorySet'
     self.outputFormat = 'HistorySet'
-    if not hasattr(self, 'pivotParameterID'):   self.pivotParameterID = 'Time'
-    if not hasattr(self, 'outputLen'):          self.outputLen = None
+    if not hasattr(self, 'pivotParameter'): self.pivotParameter = 'Time' #FIXME this assumes the ARMA model!  Dangerous assumption.
```
Comment (joshua-cogliati-inl) [inline framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py]: The line needs to be split for code standards.

**Comment by alfoa** [comment]
good to go to me

File: doc/user_manual/postprocessor.tex [hunk 2]
```diff
@@ -1271,15 +1272,24 @@ \subsubsection{Interfaced}
    \default{Maximum value of the variable with name of \xmlNode{pivotParameter}}
 \end{itemize}
 
+For example, consider history of data collected over three years in one-second increments,
+where the user wants a single \emph{typical year} extracted from the data.
+The user wants this data constructed by combining twelve equal \emph{typical month}
+segments.  In this case, the parameter \xmlNode{outputLen} should be \texttt{31536000} (the number of seconds
+in a year), while the parameter \xmlNodt{subseqLen} should be \texttt{2592000} (the number of seconds in a
```
Comment (joshua-cogliati-inl) [inline doc/user_manual/postprocessor.tex]: \xmlNodt should be \xmlNode

File: doc/user_manual/postprocessor.tex [hunk 3]
```diff
@@ -1276,7 +1276,7 @@ \subsubsection{Interfaced}
 where the user wants a single \emph{typical year} extracted from the data.
 The user wants this data constructed by combining twelve equal \emph{typical month}
 segments.  In this case, the parameter \xmlNode{outputLen} should be \texttt{31536000} (the number of seconds
-in a year), while the parameter \xmlNodt{subseqLen} should be \texttt{2592000} (the number of seconds in a
+in a year), while the parameter \xmlNode{subseqLen} should be \texttt{2592000} (the number of seconds in a
```
Comment (PaulTalbot-INL) [inline doc/user_manual/postprocessor.tex]: Yes, thank you.  Change approved.

File: framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py [hunk 4]
```diff
@@ -51,115 +69,168 @@ def run(self,inputDic):
...
+            # extract applicable information from the feature set
+            reshapedData[newHistoryCounter][feature] = np.extract(extractCondition, inputDict['output'][historyNumber][feature])
+          #increment history counter
+          newHistoryCounter += 1
+          #update new start/end points for grabbing the next history
+          startPivot = endPivot
+          endPivot += self.outputLen
+
+    inputDict['output'] = reshapedData
+    self.numHistory = len(inputDict['output'].keys()) #should be same as newHistoryCounter - 1, if that's faster
```
Comment (joshua-cogliati-inl) [inline framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py]: This doesn't need to be fixed, but sometimes comments like this make good warnings (as in if newHistoryCounter - 1 != len(...) raiseAWarning(...) )

Comment (joshua-cogliati-inl) [inline framework/PostProcessorFunctions/TypicalHistoryFromHistorySet.py]: (Or possibly good errors)

**Comment by joshua-cogliati-inl** [comment]
Merge Review Checklist
- [x] 1. Review all computer code. *Done*
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).  *No changes to input syntax*
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://hpcgitlab.inl.gov/idaholab/raven/wikis/Developer_Information#python-code-standard) for details. *Standards respected*
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass. *Automated tests pass*
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.  *No significant functionality added*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.  *No changes*
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.  *Close checklist done*
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?  *No change.*

**Pull request merged**